codec/encoder/core/src/encoder_ext.cpp

```diff
@@ -3907,6 +3907,61 @@ int32_t WelsEncoderEncodeExt (sWelsEncCtx* pCtx, SFrameBSInfo* pFbi, const SSour
         WelsLog (pLogCtx, WELS_LOG_WARNING,
                  "WelsEncoderEncodeExt()MinCr Checking,codec bitstream size is larger than Level limitation");
     }
+
+    if (iSpatialNum == 1 && NULL != pFbi->pReconPic
+        && (NULL != pFbi->pReconPic->pData[0]
```

Comment:
I would also invert these checks to return early to avoid an extra indentation:

```
if (iSpatialNum != 1 || NULL == pFbi->pReconPic || 
       (NULL == pFbi->pReconPic->pData[0] &&
        NULL == pFbi->pReconPic->pData[1] &&
        NULL == pFbi->pReconPic->pData[2]) {
     return;
 }
```